### PR TITLE
IgnoreCookie: added support for cookie_domain and ignore cookie signature

### PR DESCRIPTION
**Pull request by MichaelRoosz**
Cookie: added parameter 'validateSignature' to switch off signature validation
IgnoreCookie: added support for setting "cookie_domain"
IgnoreCookie: do not validate cookie signature to allow sharing it between multiple matomo instances

### CODE REVIEW

**Review by diosmosis — APPROVED** [review]
LGTM. I don't think we care if the ignore cookie's signature is absent or incorrect, @tsteur / @sgiehl ?

**Comment by tsteur** [comment]
@MichaelHeerklotz @diosmosis I'm actually not quite getting why something should be or should not be signed. Think I'm missing something. What and how exactly are we signing things here? For what reason? Sorry I'm not much into this.

**Comment by diosmosis** [comment]
@tsteur my guess was the ignore cookie was being created outside of matomo or something like that. But that's just a guess.

**Comment by MichaelHeerklotz** [comment]
@tsteur @diosmosis 
my guess is, that the sign check was added to avoid fake visitorids / etc created by cookie manipulation.

for the ignore cookie it makes no sense , so I have removed the check for it. this allows using the same ignore cookie for multiple matomo setups (if everything is running under the same root domain (big setups)

File: core/Cookie.php [hunk 1]
```diff
@@ -205,14 +206,15 @@ public function save($sameSite = null)
      * Extract signed content from string: content VALUE_SEPARATOR '_=' signature
      *
      * @param string $content
+     * @param bool $validate
      * @return string|bool  Content or false if unsigned
      */
-    private function extractSignedContent($content)
+    private function extractSignedContent($content, $validate = true)
```
Comment (tsteur) [inline core/Cookie.php:212]: Not too important but maybe the parameter doesn't need to have a default value?

File: core/Cookie.php [hunk 2]
```diff
@@ -227,9 +229,9 @@ private function extractSignedContent($content)
      * Unserialize the array when necessary.
      * Decode the non numeric values that were base64 encoded.
      */
-    protected function loadContentFromCookie()
+    protected function loadContentFromCookie($validateSignature = true)
```
Comment (tsteur) [inline core/Cookie.php:232]: Not too important but maybe the parameter doesn't need to have a default value?

**Review by tsteur — CHANGES_REQUESTED** [review]
@MichaelHeerklotz left a minor comment, otherwise looks all good  to merge

File: core/Cookie.php (hunk 2 above)
Comment (MichaelHeerklotz) [reply core/Cookie.php:232]: As this function is protected it maybe could break classes that extend it (some plugins maybe)? I put the default value for safety, but if you do not like it, I have nothing against removing it :)

File: core/Cookie.php (hunk 1 above)
Comment (MichaelHeerklotz) [reply core/Cookie.php:212]: As this function is private, it makes sense :+1:

Comment (tsteur) [reply core/Cookie.php:212]: if you have time be good to change otherwise not too important

**Comment by tsteur** [comment]
@MichaelHeerklotz could you resolve the merge conflict and then ping me? I'll merge it then.

**Pull request merged**